tests/services/phase.service.spec.ts

```diff
@@ -46,10 +46,11 @@ describe('PhaseService', () => {
       });
     });
 
-    it('should return an Observable of false if no openPhases are defined', () => {
+    it('should throw an error if no openPhases are defined', () => {
       githubService.fetchSettingsFile.and.returnValue(of(invalidPhasesSettingsFile));
-      phaseService.storeSessionData().subscribe((result: boolean) => {
-        expect(result).toBeFalse();
+      phaseService.storeSessionData().subscribe({
+        next: () => fail(),
+        error: (err) => expect(err).toBeInstanceOf(Error)
       });
     });
```

Comment:
Since `phaseService.storeSessionData()` uses `assertSessionDataIntegrity()` there will be an error when passing in an invalid SettingsFile. It previously didn't fail on this step because the SessionData Verification did not correctly assert that the info in the SettingsFile is correct.

Reply:
Also shall I create a separate issue to move the Session Model Test to the `test/models` directory instead of where it is currently located?

Reply:
> Since phaseService.storeSessionData() uses assertSessionDataIntegrity() there will be an error when passing in an invalid SettingsFile. It previously didn't fail on this step because the SessionData Verification did not correctly assert that the info in the SettingsFile is correct

Just to clarify the term "invalid": previously the app would have raised an error if the open phase name was invalid, but this would be detected by PhaseService. It's now caught by `assertSessionDataIntegrity`, and the error message is more specific.

Previously, we wouldn't raise an error if there was no repo defined for an open phase. We're catching this now

Reply:
> Also not sure if storeSessionData() should be used to determine if there is a valid / invalid phase like how it is used now haha.

Yea, we can remove the `return this.currentPhase !== undefined` check in `storeSessionData`, since `assertSessionDataIntegrity` is now ensuring that open phases exist and are all valid.

Reply:
> Also shall I create a separate issue to move the Session Model Test to the test/models directory instead of where it is currently located?

No need - the session model tests are in the correct location. Ideally, the `tests` directory structure should mirror the `src` directory.

In the future, we can discuss whether the unit tests should be moved into `src` (i.e. every unit's source file is stored alongside its tests). This seems to be the convention for Angular projects (but we need not necessarily adopt it).

Reply:
> Since phaseService.storeSessionData() uses assertSessionDataIntegrity() there will be an error when passing in an invalid SettingsFile. It previously didn't fail on this step because the SessionData Verification did not correctly assert that the info in the SettingsFile is correct

Just to clarify the term "invalid": previously the app would have raised an error if the open phase name was invalid, but this would be detected by PhaseService. It's now caught by `assertSessionDataIntegrity`, and the error message is more specific.

Previously, we wouldn't raise an error if there was no repo defined for an open phase. We're catching this now

Reply:
> Also not sure if storeSessionData() should be used to determine if there is a valid / invalid phase like how it is used now haha.

Yea, we can remove the `return this.currentPhase !== undefined` check in `storeSessionData`, since `assertSessionDataIntegrity` is now ensuring that open phases exist and are all valid.

Reply:
> Also shall I create a separate issue to move the Session Model Test to the test/models directory instead of where it is currently located?

No need - the session model tests are in the correct location. Ideally, the `tests` directory structure should mirror the `src` directory.

In the future, we can discuss whether the unit tests should be moved into `src` (i.e. every unit's source file is stored alongside its tests). This seems to be the convention for Angular projects (but we need not necessarily adopt it).